Sources/BraveWallet/Panels/Signature Request/SignInWithEthereumView.swift

```diff
@@ -0,0 +1,243 @@
+    if sizeCategory.isAccessibilityCategory {
+      VStack {
+        buttons
+      }
+    } else {
+      HStack {
+        buttons
+      }
+    }
+  }
+  
+  @ViewBuilder private var buttons: some View {
+    Button(action: { // cancel
+      action(false)
+    }) {
+      Text(Strings.cancelButtonTitle)
+    }
+    .buttonStyle(BraveOutlineButtonStyle(size: .large))
+    Button(action: { // approve
+      action(true)
+    }) {
+      Text(Strings.Wallet.siweSignInButtonTitle)
+    }
+    .buttonStyle(BraveFilledButtonStyle(size: .large))
+  }
+}
+
+/// The view pushed when user taps to view request details.
+private struct SignInWithEthereumDetailsView: View {
+  
+  let originInfo: BraveWallet.OriginInfo
+  let message: BraveWallet.SIWEMessage
+  
+  var body: some View {
+    ScrollView {
+      LazyVStack {
+        LazyVStack {
+          LazyVStack { // Max view count on `LazyVStack`
+            detailRow(title: Strings.Wallet.siweOriginLabel, value: Text(originInfo: originInfo))
+            Divider()
+            detailRow(title: Strings.Wallet.siweAddressLabel, value: Text(verbatim: message.address))
+            if let statement = message.statement {
+              Divider()
+              detailRow(title: Strings.Wallet.siweStatementLabel, value: Text(verbatim: statement))
+            }
+            Divider()
+            detailRow(title: Strings.Wallet.siweURILabel, value: Text(verbatim: message.uri.absoluteString))
+          }
+          LazyVStack { // Max view count on `LazyVStack`
+            Divider()
+            detailRow(title: Strings.Wallet.siweVersionLabel, value: Text(verbatim: "\(message.version)"))
+            Divider()
+            detailRow(title: Strings.Wallet.siweChainIDLabel, value: Text(verbatim: "\(message.chainId)"))
+            Divider()
+            detailRow(title: Strings.Wallet.siweIssuedAtLabel, value: Text(verbatim: message.issuedAt))
+            if let expirationTime = message.expirationTime {
+              Divider()
+              detailRow(title: Strings.Wallet.siweExpirationTimeLabel, value: Text(verbatim: expirationTime))
+            }
+            Divider()
+            detailRow(title: Strings.Wallet.siweNonceLabel, value: Text(verbatim: message.nonce))
+            if let resources = message.resources {
+              Divider()
+              detailRow(
+                title: Strings.Wallet.siweResourcesLabel,
+                value: Text(verbatim: resources.map(\.absoluteString).joined(separator: "\n"))
+              )
+            }
+          }
+        }
+        .frame(maxWidth: .infinity)
+      }
+      .padding(16)
+      .multilineTextAlignment(.leading)
+    }
+    .navigationTitle(Strings.Wallet.siweDetailsTitle)
+    .navigationBarTitleDisplayMode(.inline)
+    .background(Color(braveSystemName: .containerHighlight))
+  }
+  
+  private func detailRow(title: String, value: String) -> some View {
+    HStack(spacing: 12) {
```

Comment:
This should just be a convenience accessor to the `detailRow(title:value:)` caller below that takes a Text so you dont have to copy the entire view tree i.e.

```swift
private func detailRow(title: String, value: String) -> some View {
  detailRow(title: title, value: Text(verbatim: value))
}
```